compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -217,6 +217,24 @@ pub fn target_machine_factory(
 
     let force_emulated_tls = sess.target.force_emulated_tls;
 
+    // copy the exe path, followed by path all into one buffer
+    // null terminating them so we can use them as null terminated strings
+    let args_cstr_buff = {
```

Comment:
Can't this be CString's, so avoid playing with null bytes in rust and cpp side code? Didn't read all previous discussion, sorry, if it was suggested previously.

Reply:
I chose this version because for me it had the right balance between readability and performance for passing of CStrings through a C API. While I don't think this code is pretty, other version create more allocations, while not being much better IMO. Other views might differ and if someone feels very strongly about this I will change this.

I previously tried a version with an Vec of CStrings and a Vec with pointers to those, passing the slice of the latter to llvm::LLVMRustCreateTargetMachine.

However, that target machine factory lambda requires Fn+send+sync. I can only move the Pointer Vec into it via some terrible casting and introducing other types, because ptrs are not send+sync. If create the pointer Vec inside the lambda I have an allocation every time the lambda gets called.